builtin/providers/azure/resource_azure_dns_server.go

```diff
@@ -0,0 +1,243 @@
+			if err != nil {
+				return fmt.Errorf("Failed issuing update to network configuration: %s", err)
+			}
+			err = managementClient.WaitForOperation(reqID, nil)
+			if err != nil {
+				return fmt.Errorf("Error setting network configuration: %s", err)
+			}
+		}
+	}
+
+	// remove the resource from the state if it has been deleted in the meantime:
+	if !found {
+		d.SetId("")
+	}
+
+	return nil
+}
+
+// resourceAzureDnsServerExists does all the necessary API calls to
+// check if the DNS server definition alredy exists on Azure.
+func resourceAzureDnsServerExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	azureClient, ok := meta.(*AzureClient)
+	if !ok {
+		return false, fmt.Errorf("Failed to convert to *AzureClient, got: %T", meta)
+	}
+	managementClient := azureClient.managementClient
+	networkClient := virtualnetwork.NewClient(managementClient)
+
+	log.Println("[INFO] Fetching current network configuration from Azure.")
+	netConf, err := networkClient.GetVirtualNetworkConfiguration()
+	if err != nil {
+		return false, fmt.Errorf("Failed to get the current network configuration from Azure: %s", err)
+	}
+
+	name := d.Get("name").(string)
+
+	// search for the DNS server's definition:
+	for _, dns := range netConf.Configuration.DNS.DNSServers {
+		if dns.Name == name {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// resourceAzureDnsServerDelete does all the necessary API calls
+// to delete the DNS server definition from Azure.
+func resourceAzureDnsServerDelete(d *schema.ResourceData, meta interface{}) error {
+	azureClient, ok := meta.(*AzureClient)
+	if !ok {
+		return fmt.Errorf("Failed to convert to *AzureClient, got: %T", meta)
+	}
+	managementClient := azureClient.managementClient
+	networkClient := virtualnetwork.NewClient(managementClient)
+
+	log.Println("[INFO] Fetching current network configuration from Azure.")
+	azureClient.mutex.Lock()
+	defer azureClient.mutex.Unlock()
+	netConf, err := networkClient.GetVirtualNetworkConfiguration()
+	if err != nil {
+		return fmt.Errorf("Failed to get the current network configuration from Azure: %s", err)
+	}
+
+	name := d.Get("name").(string)
+
+	// search for the DNS server's definition and remove it:
+	for i, dns := range netConf.Configuration.DNS.DNSServers {
+		if dns.Name == name {
+			netConf.Configuration.DNS.DNSServers = append(
+				netConf.Configuration.DNS.DNSServers[:i],
+				netConf.Configuration.DNS.DNSServers[i+1:]...,
+			)
+		}
+	}
+
+	// send the configuration back to Azure:
+	log.Println("[INFO] Sending updated network configuration back to Azure.")
+	reqID, err := networkClient.SetVirtualNetworkConfiguration(netConf)
+	if err != nil {
+		return fmt.Errorf("Failed issuing update to network configuration: %s", err)
+	}
+	err = managementClient.WaitForOperation(reqID, nil)
+	if err != nil {
+		return fmt.Errorf("Error setting network configuration: %s", err)
+	}
+
+	d.SetId("")
```

Comment:
This is not needed. If the delete function succeeds without error, Terrafrom will automatically delete the resource from it's state